modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,260 @@ class LinearAlgebraError : Error {
+    return this.A8;
+  }
+
+  proc comp_A10(){
+    if !isAComputed[5] then {
+      this.A10 = matrix_prod(comp_A8(), comp_A2());
+      isAComputed[5] = true;
+    }
+    return this.A10;
+  }
+
+  proc matrix_prod(A: [], B: []){
+    return dot(A,B);
+  }
+
+  proc comp_Exactd4(){
+    if !isExactdComputed[1] then {
+      this.Exactd4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+      isExactdComputed[1] = true;
+    }
+    return this.Exactd4;
+  }
+
+  proc comp_Exactd6(){
+    if !isExactdComputed[2] then {
+      this.Exactd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+      isExactdComputed[2] = true;
+    }
+    return this.Exactd6;
+  }
+
+  proc comp_Exactd8(){
+    if !isExactdComputed[3] then {
+      this.Exactd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+      isExactdComputed[3] = true;
+    }
+    return this.Exactd8;
+  }
+
+  proc comp_Exactd10(){
+    if !isExactdComputed[4] then {
+      this.Exactd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+      isExactdComputed[4] = true;
+    }
+    return this.Exactd10;
+  }
+
+  proc comp_Loosed4(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd4();
+    }
+    else {
+      if !isApproxdComputed[1] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+        isApproxdComputed[1] = true;
+      }
+      return this.Approxd4;
+    }
+  }
+
+  proc comp_Loosed6(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd6();
+    }
+    else {
+      if !isApproxdComputed[2] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+        isApproxdComputed[2] = true;
+      }
+      return this.Approxd6;
+    }
+  }
+
+  proc comp_Loosed8(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd8();
+    }
+    else {
+      if !isApproxdComputed[3] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+        isApproxdComputed[3] = true;
+      }
+      return this.Approxd8;
+    }
+  }
+
+  proc comp_Loosed10(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd10();
+    }
+    else {
+      if !isApproxdComputed[4] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+        isApproxdComputed[4] = true;
+      }
+      return this.Approxd10;
+    }
+  }
+
+  proc pade3(){
+    var b = [120.0, 60.0, 12.0, 1.0];
```

Comment:
Somewhat stylistic -- I'd make `b` a `const`. Might allow for some optimizations, and conveys the intent more clearly.

Note that these are going to be `real(64)` by default, which will force all the results to be that type. We have a few options here -- we could start by specializing to `real(64)` (and worry about other types later, if ever!), or these need to be correctly cast. My preference is the former, but @ben-albrecht might have a different take.

Reply:
For now, If its fine, I will be going ahead with ``` real ``` data type.